src/probe/probe.c

```diff
@@ -311,14 +311,15 @@ int probe_deinit(void)
 	}
 
 	if (_probe->ext_dma.stream_tag != PROBE_DMA_INVALID) {
-		tracev_probe("probe_deinit() Freeing extraction DMA.");
+		tracev_probe("probe_deinit() Freeing task and extraction DMA.");
+		schedule_task_free(&_probe->dmap_work);
 		err = probe_dma_deinit(&_probe->ext_dma);
 		if (err < 0)
 			return err;
 	}
 
 	rfree(_probe);
-	_probe = NULL;
+	sof_get()->probe = NULL;
```

Comment:
Would it be a good idea to first set the pointer and then free? Or are we protected against racing accesses to the probe structure anyway?

Reply:
I think its safe, probes are dead at this stage but anyway why not :) Changed